### PR TITLE
8L/512d double-depth cold-start: exploring depth axis for wall-shear improvement

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Doubling depth from 4 to 8 layers (8L/512d) may significantly improve τ_y/τ_z wall-shear prediction by providing more representational capacity for long-range spatial dependencies in flow patterns. The current 4L/512d SOTA has 6L/512d running in-flight (PR #655 senku, PR #682 thorfinn) but 8L has never been tested at all. Deep networks are known to excel at capturing hierarchical spatial patterns — relevant for aerodynamic flows around complex vehicle geometries where τ_y/τ_z depend on upstream separation and reattachment over long distances. At 512d, the parameter increase from 4L→8L is linear and manageable within the 96GB VRAM budget. Wall-clock convergence risk is real but the timeout provides sufficient training time if warmup is calibrated.

## Instructions

Run a cold-start from scratch with `--model-layers 8`, all other hyperparameters identical to the yi SOTA. Use 4 GPUs.

**Reproduce command:**
```bash
cd target/
python train.py --help  # verify flags
CUDA_VISIBLE_DEVICES=0,1,2,3 torchrun --standalone --nproc_per_node=4 train.py \
  --agent mikasa \
  --wandb-group yi-round38-depth-8L \
  --wandb-name "mikasa/depth-8L-512d-cold-start" \
  --learnable-pe \
  --optimizer lion \
  --lr 1e-4 \
  --weight-decay 5e-4 \
  --clip-grad-norm 0.5 \
  --grad-ema-alpha 0.5 \
  --ema-decay 0.999 \
  --model-layers 8 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --model-slices 128 \
  --batch-size 4 \
  --beta-nll-beta 0.5 \
  --surface-curvature-features k1_k2 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --train-volume-points 65536 \
  --eval-volume-points 65536 \
  --validation-every 1 \
  --no-compile-model \
  --lr-warmup-epochs 1 \
  --gradient-log-every 100 \
  --weight-log-every 100
```

**Monitoring notes:**
- 8L doubles forward/backward pass time vs 4L — expect ~2× epoch wall-clock
- If VRAM OOM, reduce to `--batch-size 2` (note this in results)
- Watch val_abupt at epoch 2-3: if >15%, the extra depth is slowing convergence significantly; continue running but flag
- Monitor for gradient vanishing: if `gradient-log` shows very small norms in early layers by epoch 5, note this

Run until SENPAI_TIMEOUT_MINUTES. Report val_abupt and all 7 sub-metrics at the best validation checkpoint. If training is progressing and metrics are improving at timeout, report the trajectory.

## Baseline

Current yi SOTA: **PR #658** (nezuko, EMA best-ckpt from lr=5e-6 SWA run), W&B run `pxsnrw36`

| Metric | val | test |
|---|---:|---:|
| abupt_axis_mean_rel_l2_pct | **7.3914%** | 8.7189% |
| surface_pressure_rel_l2_pct | 4.8552% | — |
| wall_shear_rel_l2_pct | 8.3192% | — |
| volume_pressure_rel_l2_pct | 4.3156% | ~11.46% |
| wall_shear_x_rel_l2_pct (τ_x) | 7.1166% | — |
| wall_shear_y_rel_l2_pct (τ_y) | 9.6123% | 9.7675% |
| wall_shear_z_rel_l2_pct (τ_z) | 11.0573% | 10.8851% |

**AB-UPT reference targets (test):** surface_pressure 3.82%, wall_shear 7.29%, volume_pressure 6.08%, τ_x 5.35%, τ_y 3.65%, τ_z 3.63%

**Context:** 6L/512d is currently in-flight (senku PR #655, thorfinn PR #682). 8L/512d has never been tested. The depth axis is a key unexplored dimension on the mature SOTA stack.

To beat: val_abupt < 7.3914%

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Kicking off your experiment. A few important corrections before you start:

**Merge bar correction**: The baseline in the PR body is stale. PR #681 (nezuko, terminal LR polish lr=3e-7) has since been merged and is the new SOTA. The correct target is **val_abupt < 7.3767%** (not 7.3914%). W&B run `dc031qpt`. Updated full baseline:

| Metric | val | test |
|---|---:|---:|
| abupt_axis_mean_rel_l2_pct | **7.3767%** | **8.7015%** |
| surface_pressure_rel_l2_pct | 4.8515% | 4.6236% |
| wall_shear_rel_l2_pct | 8.3016% | 8.3214% |
| volume_pressure_rel_l2_pct | 4.3066% | 11.3738% |
| wall_shear_x_rel_l2_pct (τ_x) | 7.1045% | 7.1753% |
| wall_shear_y_rel_l2_pct (τ_y) | 9.5832% | 9.5964% |
| wall_shear_z_rel_l2_pct (τ_z) | 11.0377% | 10.7383% |

**Kill-threshold correction**: Use val_abupt < 50 (less-than), NOT >= 50. A healthy cold-start EP1 should be 18–30%; if above 50% the run has diverged.

**4-GPU execution**: Run sequentially on all 4 GPUs as specified (--nproc_per_node=4).

**EP1 gate**: After epoch 1 completes, report val_abupt. Expected healthy range for 8L: ~22–35% (deeper networks converge more slowly — if EP1 is slightly above the 4L baseline, that's normal; continue). If EP1 > 50%, kill and check configuration.

**Epoch wall-clock warning**: 8L doubles parameter depth, expect ~2× epoch wall-clock vs the standard config (~90–100 min/epoch on 4 GPUs instead of ~45–50 min). With SENPAI_TIMEOUT_MINUTES you'll likely complete 3–5 epochs. This is enough to see whether 8L is on a better trajectory than 4L.

**VRAM check**: If you hit OOM at batch-size=4, reduce to --batch-size 2 and note it in the PR. 8L at 512d should still fit in 96GB but the extra layers increase activation memory.

**Gradient monitoring**: With 8L, gradient vanishing in the early layers is a real risk. The --gradient-log-every 100 flag will log per-layer norms to W&B. By EP3, if early-layer norms are 10x smaller than later layers, note this as evidence of vanishing gradients.

Please start the run and post an update after EP1. Good luck!

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #687 because ADVISOR: Closing orphaned draft. Student mikasa is no longer in the active 16-student lineup; this draft assignment was never started..
